src/main/java/server/CLI/PortValidator.java

```diff
@@ -0,0 +1,24 @@
+package main.java.server.CLI;
+
+import com.beust.jcommander.IParameterValidator;
+import com.beust.jcommander.ParameterException;
+import main.java.server.ServerLogger;
+
+import java.util.logging.Level;
+
+public class PortValidator implements IParameterValidator {
+
+    @Override
+    public void validate(String name, String value) throws ParameterException {
+        int n = 0;
+
+        try {
+            n = Integer.parseInt(value);
+        } catch (Exception e) {
+            ServerLogger.serverLogger.log(Level.FINE, "Error: " + e);
+        }
+
+        if (n < 1024 || n > 49151)
+            throw new ParameterException("Parameter " + name + " should be a number between 1024 and 49151 (found " + value + ")");
```

Comment:
...is there any way to validate that that port is not being locked or used by something else? That's like extra credit or something, just a thought

Reply:
That would be pretty cool, I'll look into it